src/segmantic/seg/monai_unet.py

```diff
@@ -361,15 +459,27 @@ def train(
         save_top_k=3,
     )
 
+    # defining early stopping. When val loss improves less than 0 over 30 epochs, the training will be stopped.
+    early_stop_callback = EarlyStopping(monitor="val_dice",
+                                        min_delta=0.00,
+                                        patience=early_stop_patience,
+                                        mode='max',
+                                        check_finite=True,
+                                        )
+
+    lr_monitor = LearningRateMonitor(log_momentum=True,
+                                     logging_interval='epoch')
+
     print_config()
 
     # initialise Lightning's trainer.
     trainer = pl.Trainer(
         gpus=gpu_ids,
+        auto_scale_batch_size=True,
```

Comment:
does this option improve performance or training results in general?

Reply:
It might actually help with both. Like it is supposed to increase training speed and help with the gradients. I saw it when I went through this post: https://pytorch-lightning.readthedocs.io/en/latest/advanced/training_tricks.html. However, I think this is not yet actually working since there needs to be a self.batch_size in the class. Will add a ToDo comment and try playing with it to make it work properly. We will also need to consider the other things that already influence our batch size such as the num_samples and I think training data loader.